src/platform/silabs/ConnectivityManagerImpl_WIFI.cpp

```diff
@@ -262,8 +262,9 @@ void ConnectivityManagerImpl::DriveStationState()
         // If the WiFi station interface is no longer enabled, or no longer provisioned,
         // disconnect the station from the AP, unless the WiFi station mode is currently
         // under application control.
-        if (mWiFiStationMode != kWiFiStationMode_ApplicationControlled &&
-            (mWiFiStationMode != kWiFiStationMode_Enabled || !IsWiFiStationProvisioned()))
+        // if (mWiFiStationMode != kWiFiStationMode_ApplicationControlled &&
```

Comment:
Why keep the previous commented-out code?

Reply:
Removed the previous commented-out code.